intake_stac/catalog.py

```diff
@@ -83,7 +83,7 @@ def from_url(cls, url, **kwargs):
         kwargs : dict, optional
             Passed to intake.Catolog.__init__
         """
-        stac_obj = cls._stac_cls.open(url)
+        stac_obj = cls._stac_cls.from_file(url)
```

Comment:
Not sure this is useful here, but in case you didn't know there is a `pystac.read_file` that will determine the STACObjectType for you and return the right thing.